lib/Conversion/ExportVerilog/PrepareForEmission.cpp

```diff
@@ -526,10 +754,80 @@ static bool reuseExistingInOut(Operation *op) {
   return true;
 }
 
+/// Return true if this is something that will get printed as a unary operator
+/// by the Verilog printer.
+/// TODO: This is copied from PrettifyVerilog. Create a common library for
+/// verilog expression stuffs.
+static bool isVerilogUnaryOperator(Operation *op) {
+  if (isa<comb::ParityOp>(op))
+    return true;
+
+  if (auto xorOp = dyn_cast<comb::XorOp>(op))
+    return xorOp.isBinaryNot();
+
+  if (auto icmpOp = dyn_cast<comb::ICmpOp>(op))
+    return icmpOp.isEqualAllOnes() || icmpOp.isNotEqualZero();
+
+  return false;
+}
+
+/// Return true if it is beneficial to spill the operation under the specified
+/// spilling heuristic.
+bool shouldSpillWire(Operation &op, const LoweringOptions &options,
```

Comment:
Now that there are two `shouldSpillWire` that only differ in their arguments, maybe we can rename to make it more clear. It seems the original one is about spilling to wires that are required, whereas this one is just about achieving good verilog quality.

Reply:
Move the function into `EmittedExpressionStateManager::shouldSpillWireBasedOnState`.